tooling/generate-index.js

```diff
@@ -0,0 +1,39 @@
+
+const fs = require('fs');
+const path = require('path');
+
+// set directories and paths
+const methodsDir = path.join(__dirname, '../methods');
+const indexFile = path.join(methodsDir, 'index.json');
+
+// Process all method files and generate an index file
+(async () => {
```

Comment:
This doesn't appear to need to be `async` ... perhaps the `readdir`/`readFile`/`writeFile` methods were meant to be promisified below or something, but they are the `Sync` versions. The IIFE is probably unnecessary if it's not `async` as well.

Reply:
Agree, needs cleanup (but not super concerned about fixing that before merge).